src/core/device.c

```diff
@@ -111,6 +111,28 @@ static void device_done(Unit *u) {
         d->wants_property = strv_free(d->wants_property);
 }
 
+static int device_load(Unit *u) {
+        int r;
+
+        r = unit_load_fragment_and_dropin_optional(u);
+        if (r < 0)
+                return r;
+
+        if (!u->description) {
+                /* Generate a description based on the path, to be used until the
+                   device is initialized properly */
+                char *p;
+
+                r = unit_name_to_path(u->id, &p);
+                if (r < 0)
+                        return log_unit_debug_errno(u, r, "Failed to unescape name: %m");
+
+                u->description = p;
```

Comment:
why define `p` here? you can just tell unit_name_to_path() to write directly to `u->description`

Reply:
Indeed. Updated.